File: app/src/main/java/com/example/guessingnumber/MainActivity.java
package com.example.guessingnumber;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import java.util.Random;

public class MainActivity extends AppCompatActivity {

    int n;
    String message1 ="";
    int numberOfTimes = 0;

    public void randomNumberGenerator(){
        Random randomNum = new Random();
        n = randomNum.nextInt(1000) + 1;

    }

    public void guess(View view){
        EditText guess = (EditText) findViewById(R.id.guess);
        int guessInt = Integer.parseInt(guess.getText().toString());

        if(n > guessInt) {
            message1 = "Higher!!";
            numberOfTimes++;

        }
        else if(n < guessInt){
            message1 = "Lower!!";
            numberOfTimes++;
        }
        else if(n == guessInt) {
            numberOfTimes++;
            message1 = "Correct! and You guessed all " + numberOfTimes +  " times.";
            randomNumberGenerator();
            numberOfTimes = 0;
        }

        TextView txt = (TextView) findViewById(R.id.result);
        txt.setText(message1);

    }
    public void restart(View view){
        randomNumberGenerator();
        TextView txt = (TextView) findViewById(R.id.result);
        txt.setText("");
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        randomNumberGenerator();
    }
}

File: app/src/main/java/com/example/guess/MainActivity.kt
package com.example.guess

import android.content.Intent
import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import java.util.*
import android.widget.Toast;


class MainActivity : AppCompatActivity() {

    private lateinit var hint: TextView
    private lateinit var inputNumber: EditText
    private lateinit var summit: Button




    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        inputNumber = findViewById(R.id.inputNumber)
        summit = findViewById(R.id.summit)
        hint = findViewById(R.id.hint)


        randomNumbers()
    }

    private fun randomNumbers() {
        val r = Random()
        var number = r.nextInt(1000)

        var time = 0

        summit.setOnClickListener {
            var check_null = inputNumber.text.isNullOrEmpty()

            if (!check_null) {
                var checkInput = Integer.parseInt(inputNumber.text.toString())
                if (number == checkInput) {
                    var num = number.toString()
                    var input = inputNumber.text.toString()
                    time++
                    val intent = Intent(this, SecondActivity::class.java)
                    intent.putExtra("keyname", time.toString())
                    startActivity(intent)

                } else {
                    if (checkInput > number) {
                        hint.text = "Hints: Try lower"
                    } else {
                        hint.text = "Hints: Try higher"
                    }
                    time++

                }
            }else{
                val toast = Toast.makeText(this@MainActivity, "Please Enter Number", Toast.LENGTH_SHORT)
                    toast.setGravity(24, 20, 150)

                    toast.show()

            }





        }

    }

}